src/com/twitter/intellij/pants/service/project/PythonVenvFinder.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+package com.twitter.intellij.pants.service.project;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.twitter.intellij.pants.service.project.model.PythonInterpreterInfo;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class PythonVenvFinder {
```

Comment:
Will the finder find what was generated by the plugin? Is the fined needed at all if we know where we put the env, and which env we need to use?

Reply:
The venv_builder tool is invoked only if the finder did not find an existing virtual environment. After a new environment has been created, there is no point in running the finder again.

However, if there's a previously generated environment that was used for VS Code, the finder will detect it, and the venv_builder will not be invoked.